pallets/pallet-lottery/src/staking/deposit_strategies.rs

```diff
@@ -40,12 +40,45 @@ pub(super) fn reactivate_bottom_collators<T: Config>(
         let staked = StakedCollators::<T>::get(collator.clone());
         let info = pallet_parachain_staking::Pallet::<T>::candidate_info(collator.clone())
             .expect("is active collator, therefore it has collator info. qed");
+
+        // If collator not exist in delegatorState(PotAccount).delegations, ignore
+        if let Some(state) =
+            pallet_parachain_staking::Pallet::<T>::delegator_state(crate::Pallet::<T>::account_id())
+        {
+            let mut is_kick = true;
+            for x in &state.delegations.0 {
+                if x.owner == collator {
+                    is_kick = false;
+                    break;
+                }
+            }
+            if is_kick {
+                log::debug!(
+                    "Staked collator:{:?} is kicked off from delegator state, ignore.",
+                    collator
+                );
+                continue;
+            }
+        }
```

Comment:
Can we remove  from storage `StakedCollators`?

Reply:
let's just ignore the collator for now. maybe need another PR to remove it.